### PR TITLE
Add github action for unit test and coverage

### PR DESCRIPTION
**Pull request by Ashmita152**
Signed-off-by: Ashmita Bohara <ashmita.bohara152@gmail.com>

## Which problem is this PR solving?
This PR is the first step in moving from travis to github action.

## Short description of the changes
We run the unit tests and coverage job in github action. 
Sample Run: https://github.com/Ashmita152/jaeger/actions/runs/383055239

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - name: install tools
+      run: make install-ci
+
+    - name: run unit tests
+      run: make test-ci
+
+    - name: prepare coverage file
+      run: mv cover.out coverage.txt
+
+    - name: run codecov bash uploader
```
Comment (jpkrohling) [inline .github/workflows/unit-tests.yml:30]: Would it make sense to use their Github Action? 

https://github.com/codecov/codecov-action

Comment (Ashmita152) [reply .github/workflows/unit-tests.yml:30]: Sure.

**Review by jpkrohling — APPROVED** [review]
LGTM. Is there an issue where we are tracking what things we've moved already?

**Comment by Ashmita152** [comment]
Let me add a comment to this issue https://github.com/jaegertracing/jaeger/issues/2645 and make subtasks for the travis jobs to migrate.

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**